plugins/callback/mail.py

```diff
@@ -94,16 +95,21 @@ def set_options(self, task_keys=None, var_options=None, direct=None):
         self.cc = self.get_option('cc')
         self.bcc = self.get_option('bcc')
 
+    @staticmethod
+    def encode_addresses(addresses):
+        addresses_list = [email.utils.formataddr((name, address), charset='utf-8') for name, address in email.utils.getaddresses([addresses])]
+        return ','.join(addresses_list)
```

Comment:
The other `split()` calls below need also to be updated. But let's do that after #4026 has been merged, otherwise there are just more conflicts.